### PR TITLE
[GEOS-7515] JDBCStore: plugin + docs

### PR DESCRIPTION
**Pull request by NielsCharlier**
https://osgeo-org.atlassian.net/browse/GEOS-7515

### CODE REVIEW

**Comment by bencaradocdavies** [comment]
@NielsCharlier GitHub tells me there are merge conflicts. I also updated the title (7515 not 7514) and added a Jira link.

**Comment by NielsCharlier** [comment]
done

**Comment by NielsCharlier** [comment]
Eh, this failure??? Doesn't happen locally and nothing in this PR could explain it.

File: doc/en/user/source/community/jdbcstore/installing.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.. _community_jdbcconfig_installing:
```
Comment (bencaradocdavies) [inline doc/en/user/source/community/jdbcstore/installing.rst]: @NielsCharlier this should be ```_community_jdbcstore_installing```; the current label duplicates one used in ```jdbcconfig``` and causes a WARNING:
```/path/to/doc/en/user/source/community/jdbcstore/installing.rst:4: WARNING: duplicate label community_jdbcconfig_installing, other instance in /path/to/doc/en/user/source/community/jdbcconfig/installing.rst
```

File: doc/en/user/source/community/jdbcstore/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+The JDBCStore module is configured in the file ``jdbcstore/jdbcstore.properties`` inside the :ref:`datadir`.
+The following properties may be set:
+
+- ``enabled``: Use JDBCStore.  Turn off to use the data directory for all configuration instead.
+
+- ``initdb``: Initialize an empty database if this is set on true. 
+
+- ``import`` : The import configuration option tells GeoServer whether to import the current :ref:`datadir` from the file system to the database or not. If set to true, it will be imported and the config option will be set the value 'false' for the next start up to avoid trying to re-import the catalog  configuration.
+
+- ``initScript``: Path to initialisation script .sql file. Only used if initdb = true.
```
Comment (bencaradocdavies) [inline doc/en/user/source/community/jdbcstore/configuration.rst]: Perhaps change "initdb = true" to "```initdb``` is true" for consistent monospace and preferring words over symbols in running text.

File: doc/en/user/source/community/jdbcstore/configuration.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+- ``password``: JDBC connection password.
+
+- ``pool.minIdle``: minimum connections in pool  
+
+- ``pool.maxActive``:  maximum connections in pool
+
+- ``pool.poolPreparedStatements``: whether to pool prepared statements
+
+- ``pool.maxOpenPreparedStatements``: size of prepared statement cache, only used if pool.poolPreparedStatements = true
```
Comment (bencaradocdavies) [inline doc/en/user/source/community/jdbcstore/configuration.rst]: Perhaps monospace and "is" not "="?

File: doc/en/user/source/community/jdbcstore/configuration.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+- ``pool.testOnBorrow``: whether to validate connections when obtaining from the pool
+
+- ``pool.validationQuery``: validation query for connections from pool, must be set when pool.testOnBorrow = true
```
Comment (bencaradocdavies) [inline doc/en/user/source/community/jdbcstore/configuration.rst]: Perhaps monospace and "is" not "="?

**Comment by bencaradocdavies** [comment]
I have confirmed that a -PcommunityRelease and community assembly builds the -jdbcstore-plugin.zip as expected.

@NielsCharlier this looks good; the documentation is comprehensive, and other than the line comment issues (including one duplicate label that should be fixed), this is ready to merge.

**Comment by bencaradocdavies** [comment]
A full build with ```-Prelease``` shows no sign of the build failure reported by Travis CI. I think the build failure seen above is a race condition. The NPE occurs at FileSystemResourceTheoryTest:157, which is the last line here:
```
listener.reset();
fileA.createNewFile();
n = listener.await(5,TimeUnit.SECONDS);
assertEquals(Kind.ENTRY_MODIFY,n.getKind());
```
See: https://github.com/geoserver/geoserver/blob/master/src/platform/src/test/java/org/geoserver/platform/resource/FileSystemResourceTheoryTest.java#L157

Tests with timed waits are vulnerable to race conditions on virtual machines where execution or I/O might be paused for long periods, or time changes might be irregular I expect that the Jenkins CI failure was cause by virtual machine time irregularity.

**Comment by NielsCharlier** [comment]
Thanks for the review, Ben

**Comment by tbarsballe** [comment]
Backported to 2.9.x

**Pull request merged**